### PR TITLE
Test bug #154.

### LINKED ISSUE #154: [8.0] partner_relations - Error when deleting a res.partner
Tested in OCA's runbot
https://runbot.odoo-community.org/runbot/repo/github-com-oca-partner-contact-134
- Create a partner
- Don't select "is a company"
- Filled "firstname" and "lastname"
- Save partner
- Delete the partner we have created
- You get this error:

> AccessError
> No value found for res.partner(75,).birthdate_date

I have tested in Odoo's runbot and you can delete any res.partner if it hasn't any other relation with other objects.

Maybe this bug it not from this module partner_contact_birthdate

@pedrobaeza I think that any module in this project must at least pass test of creating, modifying and deleting a res.partner, if not shouldn't be merged.

cc @sysadminmatmoz @Yajo

### PR DESCRIPTION
**Pull request by yajo**
I still do not understand the cause of this issue. Normal behavior is that the modified line raises a `MissingError` and it works fine. However installing `partner_relations` makes it to raise an `AccessError` which does not get handled and prevents unlinking.

This just checks the existence of the partner before computing it, and skips if false.

This adds also a test case. Merge it to fix #154.

### CODE REVIEW

File: partner_firstname/__openerp__.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 {
     'name': 'Partner first name and last name',
     'summary': "Split first name and last name for non company partners",
-    'version': '2.0',
+    'version': '2.1.0',
```
Comment (pedrobaeza) [inline partner_firstname/__openerp__.py]: Use full version numbering 8.0.2.1.0

**Comment by eLBati** [comment]
Please see travis errors

**Comment by yajo** [comment]
[`test_flake8`](https://travis-ci.org/OCA/partner-contact/jobs/79234878#L181) could be more descriptive... Anyway I think it's fixed now.

I'd like to know what is `partner_relations` doing to break these modules. :confused:

**Comment by hbrunn** [comment]
@Yajo see #171. I'm not sure if you still want to include the tests? They don't hurt IMHO

**Comment by yajo** [comment]
OK.

**Comment by yajo** [comment]
Removed workaround and leaving tests.

**Comment by pedrobaeza** [comment]
Please validate then the other PR to merge it to get this one green

**Comment by yajo** [comment]
This is weird, because after rebasing this morning tests pass here, but they are really failing if you install `partner_relations`. I tried locally, but you can do the manual test of deleting a partner in the runbot and you will see.

**Comment by yajo** [comment]
Tests still fail in local machine.

**Comment by rafaelbn** [comment]
@eLBati @pedrobaeza @hbrunn This PR only add a TEST, I think we should merge it.
Is it OK if I merge it? Please let me know, thanks

**Comment by pedrobaeza** [comment]
But the thing that tests is already fixed?

**Comment by yajo** [comment]
Not, AFAIK.

**Comment by eLBati** [comment]
@Yajo `This branch has conflicts that must be resolved`, please rebase

**Comment by yajo** [comment]
Done.

**Comment by eLBati** [comment]
@Yajo , as @pedrobaeza was saying, if the bug tested by this PR is not fixed, why is this PR green?

**Comment by yajo** [comment]
It's because [`partner_contact_birthdate`](https://travis-ci.org/OCA/partner-contact/jobs/85065477#L215) and [`partner_firstname`](https://travis-ci.org/OCA/partner-contact/jobs/85065477#L329) get installed and tested before [`partner_relations`](https://travis-ci.org/OCA/partner-contact/jobs/85065477#L417), which is the buggy module. I suppose it's a matter of alphabetic order of module installation. Repeat tests after installing `partner_relations` and you will see the real result.

**Comment by rafaelbn** [comment]
@Yajo this is a long history. Could you please recapitulate which is the bug and in which PR it is solve? Thanks

**Comment by rafaelbn** [comment]
Please rebase

**Comment by yajo** [comment]
Rebased.

The bug is https://github.com/OCA/partner-contact/issues/154 and it is in module `partner_relations`, but it is a _conflict bug_, so that module seems to work fine, but breaks other ones like `partner_contact_birthdate`, `partner_event` and `partner_firstname` when installed together.

This PR **does not fix it, it just tests it**, but tests go into some good modules. Somebody should fix and test `partner_relations` itself (the bad module). https://github.com/OCA/partner-contact/pull/171 was supposed to be the fix, but these tests show it is not.

Besides, Travis says all is OK, but https://github.com/OCA/partner-contact/pull/170#issuecomment-147684039 explains why. I suppose some custom line to `.travis.yml` should work, but I don't know where to find docs about it.

**Comment by rafaelbn** [comment]
Please @Yajo could you resolve code conflicts please? Thanks

**Comment by rafaelbn** [comment]
Sorry to ping. As PSC Members I ask for technical opinion and help to @eLBati @tafaRU @jgrandguillaume @eLBati @pedrobaeza and because they tried to fix this issue @hbrunn 's opinion is very welcome.
- Issue: https://github.com/OCA/partner-contact/issues/154
- @yajo recapitulate https://github.com/OCA/partner-contact/pull/170#issuecomment-152111008
- Could we merge this PR to include tests?
  Thanks in advance.

**Comment by yajo** [comment]
I'm not sure merging a PR that includes failing tests is a good idea, they should be fixed before. Otherwise subsequent PRs will all fail.

**Comment by eLBati** [comment]
I think the priority is to fix https://github.com/OCA/partner-contact/issues/154
Some days ago I investigated it, but did not find solutions.
It would need deep debugging and it could also lead to an odoo core bug.
Maybe in the next weeks I will find some hours to dedicate to it.

**Comment by yajo** [comment]
I doubt it is core, since everything works without that module...

**Comment by pedrobaeza** [comment]
A rebase is needed.

**Comment by rafaelbn** [comment]
@Yajo please, could you rebase this PR? Thanks

**Comment by yajo** [comment]
Closing this, see https://github.com/OCA/partner-contact/pull/200#issuecomment-157680641.

**Pull request closed without merging**